calicoctl/commands/resources.go

```diff
@@ -267,24 +208,28 @@ func executeConfigCommand(args map[string]interface{}, action action) commandRes
 
 // executeResourceAction fans out the specific resource action to the appropriate method
 // on the ResourceManager for the specific resource.
-func executeResourceAction(args map[string]interface{}, client client.Interface, resource runtime.Object, action action) ([]runtime.Object, error) {
-	rm := resourcemgr.GetResourceManager(resource)
-	var err error
+func executeResourceAction(args map[string]interface{}, client client.Interface, res runtime.Object, action action) ([]runtime.Object, error) {
+	rm := resourcemgr.GetResourceManager(res)
+
+	resource, namespace, allNs, err := handleNamespace(res, args)
+	if err != nil {
+		return nil, err
+	}
+
 	var resOut runtime.Object
 
 	switch action {
 	case actionApply:
-		resOut, err = rm.Apply(client, resource.(resourcemgr.ResourceObject))
+		resOut, err = rm.Apply(client, resource, namespace)
```

Comment:
If we insist that the namespace is now included in the resource when required, then the namespace parameter in these signatures seems redundant.

Reply:
even in the GetOrList - provided you leave either the name or the namespace blank then the GetOrList function can do it's thing and workout what to include in the filter options.